test/e2e/storage/testsuites/provisioning.go

```diff
@@ -527,6 +528,266 @@ func (p *provisioningTestSuite) DefineTests(driver storageframework.TestDriver,
 		l.testCase.TestDynamicProvisioning(ctx)
 	})
 
+	ginkgo.It("should provision storage with pvc data source for readonly filesystem", func(ctx context.Context) {
+		if !dInfo.Capabilities[storageframework.CapPVCDataSource] {
+			e2eskipper.Skipf("Driver %q does not support cloning - skipping", dInfo.Name)
+		}
+
+		if !dInfo.Capabilities[storageframework.CapReadOnlyMany] {
+			e2eskipper.Skipf("Driver %q does not support ROX access mode - skipping", dInfo.Name)
+		}
+
+		init(ctx)
+
+		if l.config.ClientNodeSelection.Name == "" {
+			// Schedule all pods to the same topology segment (e.g. a cloud availability zone), some
+			// drivers don't support cloning across them.
+			if err := ensureTopologyRequirements(ctx, &l.config.ClientNodeSelection, l.cs, dInfo, 1); err != nil {
+				framework.Failf("Error setting topology requirements: %v", err)
+			}
+		}
+		testConfig := storageframework.ConvertTestConfig(l.config)
+		expectedContent := fmt.Sprintf("Hello from namespace %s", f.Namespace.Name)
+
+		l.sourcePVC.Name = "source-pvc"
+		dataSourceRef := preparePVCDataSourceForProvisioning(ctx, f, testConfig, l.cs, l.sourcePVC, l.sc, pattern.VolMode, expectedContent)
+		l.pvc.Spec.DataSourceRef = dataSourceRef
+		l.testCase.NodeSelection = testConfig.ClientNodeSelection
+
+		// Cloning fails if the source disk is still in the process of detaching, so we wait for the VolumeAttachment to be removed before cloning.
+		volumeAttachment := e2evolume.GetVolumeAttachmentName(ctx, f.ClientSet, testConfig, l.testCase.Provisioner, dataSourceRef.Name, l.sourcePVC.Namespace)
```

Comment:
Are all ROX volume types going to be attachable? Does the code fails gracefully when it is not attachable?

Reply:
We have a capability for this in storageframework.CapReadOnlyMany if that's what you mean? So this test is only scoped to PD and hostpath driver which do support ROX

Reply:
The storageframework is used by many other plugins outside of k/k. So from a general standpoint, we cannot assume that drivers that do support ROX also support attachments.

I think in this case though, the volumeattachment object would not exist so it would pass?

Reply:
Yeah it does seem okay at least for now. Ideally, it should be behind a capability.

Reply:
Yup, other tests in this file also test volume attachments in the same fashion we are doing here. If we want to put all of those tests behind a shared capability I think we should do that in another PR.